File: server/api/hello.ts
import type { VercelRequest, VercelResponse } from "@vercel/node";

export default async (request: VercelRequest, response: VercelResponse) => {
  return response.status(200).json({
    message: "Hello from GitHub Project Status Viewer API",
    timestamp: new Date().toISOString(),
  });
};

File: server/api/hello.go
package handler

import (
	"encoding/json"
	"net/http"
	"time"
)

type Response struct {
	Message   string `json:"message"`
	Timestamp string `json:"timestamp"`
}

func Handler(w http.ResponseWriter, r *http.Request) {
	w.Header().Set("Content-Type", "application/json")

	response := Response{
		Message:   "Hello from GitHub Project Status Viewer Go API",
		Timestamp: time.Now().Format(time.RFC3339),
	}

	w.WriteHeader(http.StatusOK)
	json.NewEncoder(w).Encode(response)
}
